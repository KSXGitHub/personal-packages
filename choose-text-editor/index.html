<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>@khai96x/choose-text-editor — Reference</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.js" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">@khai96x/choose-text-editor — Reference</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1>@khai96x/choose-text-editor — Reference</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<div class="tsd-panel tsd-typography">
				<a href="#choose-text-editor" id="choose-text-editor" style="color: inherit; text-decoration: none;">
					<h1>Choose Text Editor</h1>
				</a>
				<p>Get path to a text editor executable base on some environment variables</p>
				<a href="#configuration" id="configuration" style="color: inherit; text-decoration: none;">
					<h2>Configuration</h2>
				</a>
				<a href="#location" id="location" style="color: inherit; text-decoration: none;">
					<h3>Location</h3>
				</a>
				<p>This package uses <a href="https://github.com/davidtheclark/cosmiconfig">cosmiconfig</a> to load configuration. It will search files with matching name from working directory to <code>stopDir</code> (default to home directory).</p>
				<p><strong>Names:</strong></p>
				<ul>
					<li><code>choose-text-editor</code> (yaml format)</li>
					<li><code>choose-text-editor.json</code></li>
					<li><code>choose-text-editor.yaml</code></li>
					<li><code>choose-text-editor.config.js</code></li>
					<li><code>choose-text-editor.js</code></li>
					<li><code>package.json#choose-text-editor</code></li>
					<li><code>package.yaml#choose-text-editor</code></li>
				</ul>
				<a href="#schema" id="schema" style="color: inherit; text-decoration: none;">
					<h2>Schema</h2>
				</a>
				<p>A config file have a structure satisfies <a href="https://raw.githubusercontent.com/KSXGitHub/personal-packages/%40khai96x/choose-text-editor/0.2.1/packages/choose-text-editor/schemas/editor-set.schema.json">schemas/editor-set.schema.json</a> or <a href="https://github.com/KSXGitHub/personal-packages/blob/%40khai96x/choose-text-editor/0.2.1/packages/choose-text-editor/lib/editors.ts#L1-L5">EditorSet interface</a></p>
				<a href="#example-configuration" id="example-configuration" style="color: inherit; text-decoration: none;">
					<h2>Example Configuration</h2>
				</a>
				<pre><code class="language-json">{
  <span class="hljs-attr">"$schema"</span>: <span class="hljs-string">"https://raw.githubusercontent.com/KSXGitHub/personal-packages/%40khai96x/choose-text-editor/0.2.1/packages/choose-text-editor/schemas/editor-set.schema.json#"</span>,
  <span class="hljs-attr">"graphical"</span>: [
    { <span class="hljs-attr">"program"</span>: <span class="hljs-string">"code"</span>, <span class="hljs-attr">"flags"</span>: [<span class="hljs-string">"wait"</span>] },
    { <span class="hljs-attr">"program"</span>: <span class="hljs-string">"atom"</span>, <span class="hljs-attr">"suffixes"</span>: [<span class="hljs-string">"--wait"</span>] }
  ],
  <span class="hljs-attr">"terminal"</span>: [
    { <span class="hljs-attr">"program"</span>: <span class="hljs-string">"vim"</span> }
  ],
  <span class="hljs-attr">"chooser"</span>: <span class="hljs-string">"@khai96x/choose-text-editor@^0.2.1"</span>
}</code></pre>
				<a href="#environment-variables" id="environment-variables" style="color: inherit; text-decoration: none;">
					<h2>Environment Variables</h2>
				</a>
				<ul>
					<li>When <code>FORCE_EDITOR</code> is set, <code>choose-text-editor</code> will skip reading config and choose forced editor.</li>
					<li>When <code>FORCE_EDITOR</code> is not set, <code>ISINTTY</code> must always be set to either <code>true</code> or <code>false</code> (case-sensitive):<ul>
							<li><code>ISINTTY=true</code> to consider only terminal editors.</li>
							<li><code>ISINTTY=false</code> to consider graphical editors before terminal editors.</li>
						</ul>
					</li>
					<li><code>FORCE_EDITOR_PREFIXES</code> when set, must be a valid yaml array of string. This prefix will be added to output command line.</li>
				</ul>
				<a href="#basic-commands" id="basic-commands" style="color: inherit; text-decoration: none;">
					<h2>Basic Commands</h2>
				</a>
				<a href="#find-an-editor-and-display-its-command" id="find-an-editor-and-display-its-command" style="color: inherit; text-decoration: none;">
					<h3>Find an editor and display its command</h3>
				</a>
				<pre><code class="language-sh">ISINTTY=<span class="hljs-literal">false</span>
choose-text-editor</code></pre>
				<a href="#find-an-editor-to-open-a-file" id="find-an-editor-to-open-a-file" style="color: inherit; text-decoration: none;">
					<h3>Find an editor to open a file</h3>
				</a>
				<pre><code class="language-sh">ISINTTY=<span class="hljs-literal">false</span>
choose-text-editor -x <span class="hljs-built_in">exec</span> -- my-file.txt</code></pre>
				<a href="#print-help-message" id="print-help-message" style="color: inherit; text-decoration: none;">
					<h3>Print help message</h3>
				</a>
				<pre><code class="language-sh">choose-text-editor --<span class="hljs-built_in">help</span></code></pre>
				<a href="#license" id="license" style="color: inherit; text-decoration: none;">
					<h2>License</h2>
				</a>
				<p><a href="https://git.io/fj9XO">MIT</a> © <a href="https://github.com/KSXGitHub">Hoàng Văn Khải</a></p>
			</div>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-module"><span class="tsd-kind-icon">Module</span></li>
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-index-signature"><span class="tsd-kind-icon">Index signature</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
				<li class="tsd-kind-enum-member"><span class="tsd-kind-icon">Enumeration member</span></li>
				<li class="tsd-kind-property tsd-parent-kind-enum"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-enum"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-interface"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class"><span class="tsd-kind-icon">Accessor</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-class"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-call-signature tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
<script>if (location.protocol == 'file:') document.write('<script src="assets/js/search.js"><' + '/script>');</script>
</body>
</html>