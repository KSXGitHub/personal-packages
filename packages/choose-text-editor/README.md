# Choose Text Editor

Get path to a text editor executable base on some environment variables

## Configuration

### Location

This package uses [cosmiconfig](https://github.com/davidtheclark/cosmiconfig) to load configuration. It will search files with matching name from working directory to `stopDir` (default to home directory).

**Names:**

- `choose-text-editor` (yaml format)
- `choose-text-editor.json`
- `choose-text-editor.yaml`
- `choose-text-editor.config.js`
- `choose-text-editor.js`
- `package.json#choose-text-editor`
- `package.yaml#choose-text-editor`

## Schema

A config file have a structure satisfies [schemas/editor-set.schema.json](https://raw.githubusercontent.com/KSXGitHub/personal-packages/%40khai96x/choose-text-editor/0.1.0/packages/choose-text-editor/schemas/editor-set.schema.json) or [EditorSet interface](https://github.com/KSXGitHub/personal-packages/blob/%40khai96x/choose-text-editor/0.1.0/packages/choose-text-editor/lib/editors.ts#L1-L5)

## Example Configuration

```json
{
  "$schema": "https://raw.githubusercontent.com/KSXGitHub/personal-packages/%40khai96x/choose-text-editor/0.1.0/packages/choose-text-editor/schemas/editor-set.schema.json#",
  "graphical": [
    { "program": "code", "flags": ["wait"] },
    { "program": "atom", "suffixes": ["--wait"] }
  ],
  "terminal": [
    { "program": "vim" }
  ],
  "chooser": "@khai96x/choose-text-editor@^0.1.0"
}
```

## Environment Variables

* When `FORCE_EDITOR` is set, `choose-text-editor` will skip reading config and choose forced editor.
* When `FORCE_EDITOR` is not set, `ISINTTY` must always be set to either `true` or `false` (case-sensitive):
  - `ISINTTY=true` to consider only terminal editors.
  - `ISINTTY=false` to consider graphical editors before terminal editors.
* `FORCE_EDITOR_PREFIXES` when set, must be a valid yaml array of string. This prefix will be added to output command line.

## Basic Commands

### Find an editor and display its command

```sh
ISINTTY=false
choose-text-editor
```

### Find an editor to open a file

```sh
ISINTTY=false
choose-text-editor -x exec -- my-file.txt
```

### Print help message

```sh
choose-text-editor --help
```

## License

[MIT](https://git.io/fj9XO) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
