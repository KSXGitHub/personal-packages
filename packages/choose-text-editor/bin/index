#! /usr/bin/env node
require('process').argv[0] = 'choose-text-editor'
require('./main')
