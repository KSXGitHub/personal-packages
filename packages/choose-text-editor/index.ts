export * from './lib/choose'
export * from './lib/choose-result'
export * from './lib/clear-cache'
export * from './lib/command'
export * from './lib/console'
export * from './lib/constants'
export * from './lib/cosmiconfig'
export * from './lib/editors'
export * from './lib/encode-cli-flag'
export * from './lib/encode-cli-option'
export * from './lib/handle-choose-error'
export * from './lib/handle-chooser-validation'
export * from './lib/handle-chosen-command'
export * from './lib/json-schema'
export * from './lib/log-schema-errors'
export * from './lib/main'
export * from './lib/process'
export * from './lib/schemas'
export * from './lib/show-status'
export * from './lib/spawn-sync'
export * from './lib/status'
export * from './lib/str-to-bool'
export * from './lib/validate'
export * from './lib/which'
