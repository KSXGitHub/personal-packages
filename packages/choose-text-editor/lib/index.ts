export * from './choose'
export * from './choose-result'
export * from './clear-cache'
export * from './command'
export * from './console'
export * from './constants'
export * from './cosmiconfig'
export * from './editors'
export * from './encode-cli-flag'
export * from './encode-cli-option'
export * from './handle-chosen-command'
export * from './json-schema'
export * from './log-schema-errors'
export * from './main'
export * from './process'
export * from './schemas'
export * from './show-status'
export * from './spawn-sync'
export * from './status'
export * from './str-to-bool'
export * from './validate'
export * from './which'
