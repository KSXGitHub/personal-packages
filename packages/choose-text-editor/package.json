{
  "name": "@khai96x/choose-text-editor",
  "version": "0.0.0",
  "description": "Get path to a text editor executable base on some environment variables",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/KSXGitHub/personal-packages#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/KSXGitHub/personal-packages.git"
  },
  "bugs": {
    "url": "https://github.com/KSXGitHub/personal-packages/issues"
  },
  "sideEffects": false,
  "bin": {
    "choose-text-editor": "bin/index"
  },
  "dependencies": {
    "iter-tools": "^6.2.6",
    "which": "^2.0.1",
    "semver": "^7.1.1",
    "js-yaml": "^3.13.1",
    "parse-package-name": "^0.1.0",
    "shell-escape": "^0.2.0",
    "jsonschema": "^1.2.5",
    "cosmiconfig": "6.0.0",
    "yargs": "^15.0.2",
    "exec-once": "0.1.2",
    "string-template-format": "^0.2.5",
    "@tsfun/option": "^0.0.10",
    "@tsfun/result": "^0.0.11",
    "tslib": "^1.10.0",
    "@types/which": "^1.3.2",
    "@types/semver": "^6.2.0",
    "@types/js-yaml": "^3.12.1",
    "@types/parse-package-name": "^0.1.0",
    "@types/shell-escape": "^0.2.0",
    "@types/yargs": "^13.0.3",
    "@types/node": "^12.12.21"
  }
}
