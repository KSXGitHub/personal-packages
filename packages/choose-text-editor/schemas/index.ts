import { SchemaLoader } from '../lib'
const loader = SchemaLoader(require)
export const EditorSet = loader('./editor-set.schema.json')
