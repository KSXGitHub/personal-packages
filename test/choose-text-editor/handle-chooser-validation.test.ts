import { handleChooserValidation } from '@khai96x/choose-text-editor'
import MockedLogger from './lib/mocked-logger'

function setup(chooser: string) {
  const { log, getLogs, getText } = new MockedLogger()
  const result = handleChooserValidation(
    log,
    chooser,
    '@khai96x/choose-text-editor', // not necessary actual package name
    '3.2.1', // not necessary actual package version
    '/path/to/config/file',
  )
  return { chooser, log, getLogs, getText, result }
}

describe('when chooser is valid', () => {
  const chooser = '@khai96x/choose-text-editor@^3.2.1'

  it('does not call logError', () => {
    expect(setup(chooser).log).not.toBeCalled()
  })

  it('does not print anything', () => {
    expect(setup(chooser).getText().trim()).toBe('')
  })

  it('returns true', () => {
    expect(setup(chooser).result).toBe(true)
  })
})

describe('when chooser has invalid package name', () => {
  const chooser = 'invalid-package-name@^3.2.1'

  it('calls logError', () => {
    expect(setup(chooser).getLogs()).toMatchSnapshot()
  })

  it('prints error messages', () => {
    expect(setup(chooser).getText()).toMatchSnapshot()
  })

  it('returns false', () => {
    expect(setup(chooser).result).toBe(false)
  })
})

describe('when chooser has invalid version range', () => {
  const chooser = '@khai96x/choose-text-editor@invalid-version-range'

  it('calls logError', () => {
    expect(setup(chooser).getLogs()).toMatchSnapshot()
  })

  it('prints error messages', () => {
    expect(setup(chooser).getText()).toMatchSnapshot()
  })

  it('returns false', () => {
    expect(setup(chooser).result).toBe(false)
  })
})

describe('when chooser has a path', () => {
  const chooser = '@khai96x/choose-text-editor/lib/index.js@^3.2.1'

  it('calls logError', () => {
    expect(setup(chooser).getLogs()).toMatchSnapshot()
  })

  it('prints error messages', () => {
    expect(setup(chooser).getText()).toMatchSnapshot()
  })

  it('returns false', () => {
    expect(setup(chooser).result).toBe(false)
  })
})

describe('when chooser version is not satisfied', () => {
  const chooser = '@khai96x/choose-text-editor@>3.2.1'

  it('calls logError', () => {
    expect(setup(chooser).getLogs()).toMatchSnapshot()
  })

  it('prints error messages', () => {
    expect(setup(chooser).getText()).toMatchSnapshot()
  })

  it('returns false', () => {
    expect(setup(chooser).result).toBe(false)
  })
})
