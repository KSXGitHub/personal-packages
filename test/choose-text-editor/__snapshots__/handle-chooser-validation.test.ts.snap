// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when chooser has a path calls logError 1`] = `
Array [
  Array [
    "[ERROR] Package path is expected to NOT be specified, but it was",
  ],
  Array [
    "* config path: 'lib/index.js'",
  ],
]
`;

exports[`when chooser has a path prints error messages 1`] = `
"
[ERROR] Package path is expected to NOT be specified, but it was
* config path: 'lib/index.js'
"
`;

exports[`when chooser has invalid package name calls logError 1`] = `
Array [
  Array [
    "[ERROR] Invalid chooser",
  ],
  Array [
    "* config package: 'invalid-package-name'",
  ],
  Array [
    "* used package: '@khai96x/choose-text-editor'",
  ],
]
`;

exports[`when chooser has invalid package name prints error messages 1`] = `
"
[ERROR] Invalid chooser
* config package: 'invalid-package-name'
* used package: '@khai96x/choose-text-editor'
"
`;

exports[`when chooser has invalid version range calls logError 1`] = `
Array [
  Array [
    "[ERROR] Invalid version range for chooser",
  ],
  Array [
    "help: Read https://docs.npmjs.com/misc/semver#ranges for valid version range syntax",
  ],
  Array [
    "* config version range: 'invalid-version-range'",
  ],
]
`;

exports[`when chooser has invalid version range prints error messages 1`] = `
"
[ERROR] Invalid version range for chooser
help: Read https://docs.npmjs.com/misc/semver#ranges for valid version range syntax
* config version range: 'invalid-version-range'
"
`;

exports[`when chooser version is not satisfied calls logError 1`] = `
Array [
  Array [
    "[ERROR] Incompatible chooser",
  ],
  Array [
    "help: This version of @khai96x/choose-text-editor does not satisfied what is required in config",
  ],
  Array [
    "help: Please update @khai96x/choose-text-editor or your config",
  ],
  Array [
    "* config version: '>3.2.1'",
  ],
  Array [
    "* used version: '3.2.1'",
  ],
]
`;

exports[`when chooser version is not satisfied prints error messages 1`] = `
"
[ERROR] Incompatible chooser
help: This version of @khai96x/choose-text-editor does not satisfied what is required in config
help: Please update @khai96x/choose-text-editor or your config
* config version: '>3.2.1'
* used version: '3.2.1'
"
`;
